gluonnlp/model/utils.py

```diff
@@ -19,21 +19,25 @@
 """Building blocks and utility for models."""
 __all__ = ['apply_weight_drop']
 
+import collections
 import functools
+import re
+import warnings
 
-from mxnet.gluon import Block, HybridBlock, contrib, rnn
+from mxnet.gluon import Block, contrib, rnn
 from .parameter import WeightDropParameter
 
-def apply_weight_drop(block, local_param_name, rate, axes=(),
+# pylint: disable=too-many-nested-blocks
+def apply_weight_drop(block, local_param_regex, rate, axes=(),
                       weight_dropout_mode='training'):
     """Apply weight drop to the parameter of a block.
 
     Parameters
     ----------
     block : Block or HybridBlock
         The block whose parameter is to be applied weight-drop.
-    local_param_name : str
-        The parameter name used on the block. such as 'weight'.
+    local_param_regex : str
+        The regex for parameter names used in the self.params.get(), such as 'weight'.
```

Comment:
Can we update the doc with an example of using regex instead of exact match? such as ".*h2h_weight"